cmd/opentelemetry/app/exporter/elasticsearchexporter/estranslator/modeltranslator_test.go

```diff
@@ -0,0 +1,232 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package estranslator
+
+import (
+	"encoding/binary"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/dbmodel"
+)
+
+var (
+	traceID = []byte("0123456789abcdef")
+	spanID  = []byte("01234567")
+)
+
+func TestAttributeToKeyValue(t *testing.T) {
+	tests := []struct {
+		key      string
+		attr     pdata.AttributeValue
+		keyValue dbmodel.KeyValue
+	}{
+		{
+			key:      "foo",
+			attr:     pdata.NewAttributeValueString("bar"),
+			keyValue: dbmodel.KeyValue{Key: "foo", Value: "bar", Type: dbmodel.StringType},
+		},
+		{
+			key:      "foo",
+			attr:     pdata.NewAttributeValueBool(true),
+			keyValue: dbmodel.KeyValue{Key: "foo", Value: "true", Type: dbmodel.BoolType},
+		},
+		{
+			key:      "foo",
+			attr:     pdata.NewAttributeValueBool(false),
+			keyValue: dbmodel.KeyValue{Key: "foo", Value: "false", Type: dbmodel.BoolType},
+		},
+		{
+			key:      "foo",
+			attr:     pdata.NewAttributeValueInt(15),
+			keyValue: dbmodel.KeyValue{Key: "foo", Value: "15", Type: dbmodel.Int64Type},
+		},
+		{
+			key:      "foo",
+			attr:     pdata.NewAttributeValueDouble(16.42),
+			keyValue: dbmodel.KeyValue{Key: "foo", Value: "16.42", Type: dbmodel.Float64Type},
+		},
+	}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s:%v", test.keyValue.Key, test.keyValue.Value), func(t *testing.T) {
+			keyValue := attributeToKeyValue(test.key, test.attr)
+			assert.Equal(t, test.keyValue, keyValue)
+		})
+	}
+}
+
+func TestTagMapValue(t *testing.T) {
+	tests := []struct {
+		attr  pdata.AttributeValue
+		value interface{}
+	}{
+		{
+			attr:  pdata.NewAttributeValueString("foo"),
+			value: "foo",
+		},
+		{
+			attr:  pdata.NewAttributeValueBool(true),
+			value: true,
+		},
+		{
+			attr:  pdata.NewAttributeValueInt(15),
+			value: int64(15),
+		},
+		{
+			attr:  pdata.NewAttributeValueDouble(123.66),
+			value: float64(123.66),
+		},
+	}
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%v", test.value), func(t *testing.T) {
+			val := attributeValueToInterface(test.attr)
+			assert.Equal(t, test.value, val)
+		})
+	}
+}
+
+func TestConvertSpan(t *testing.T) {
+	traces := traces("myservice")
+	resource := traces.ResourceSpans().At(0).Resource()
+	resource.Attributes().InsertDouble("num", 16.66)
+	span := addSpan(traces, "root", traceID, spanID)
+	span.SetKind(pdata.SpanKindCLIENT)
+	span.Status().InitEmpty()
+	span.Status().SetCode(2)
```

Comment:
int `2` maps to `Status_UnknownError`. It would be better to consume the constant from the `pdata` but these constants are defined in the private package.